src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionProvider.cs

```diff
@@ -29,11 +29,11 @@ internal class GenerateMethodCodeActionProvider : IRazorCodeActionProvider
             return s_emptyResult;
         }
 
-        var change = new SourceChange(context.Location.AbsoluteIndex, length: 0, newText: string.Empty);
         var syntaxTree = context.CodeDocument.GetSyntaxTree();
-        var owner = syntaxTree.Root.LocateOwner(change);
+        var owner = syntaxTree.Root.FindToken(context.Location.AbsoluteIndex).Parent;
```

Comment:
Would love a comment here about why the `Parent` call is needed. I would guess that FindToken is returning the attribute value, and its moving up to attribute itself, but that doesn't seem to match with what the code then does with it.

Reply:
`FindToken` finds a _token_, while `FindOwner` would return full nodes. The code following this is looking for `CSharpExpressionLiteral` or `MarkupTextLiteral` nodes, which are not `SyntaxToken`s themselves, but nodes composed of `SyntaxToken`s.

Reply:
🤦🏼‍♂️

I feel like I've asked that before, and suspect I will forget again. Seems I think exclusively in nodes :)